core/src/calleeFunctions/UniswapV3Callee.ts

```diff
@@ -40,23 +39,28 @@ const getMarketPrice = async function (
     collateral: CollateralConfig,
     marketId: string,
     collateralAmount: BigNumber
-): Promise<BigNumber> {
-    // convert collateral into DAI
-    const { route, fees } = await getRouteAndGasQuote(network, collateral.symbol, collateralAmount, marketId);
-    if (!route) {
-        throw new Error(`No route found for ${collateral.symbol} to DAI`);
-    }
-    const pools = await routeToPool(network, route, collateral.symbol, fees);
-    const daiAmount = await convertCollateralToDaiUsingPool(
+): Promise<{ price: BigNumber; pools: Pool[] }> {
+    const { route, fees, totalPrice, pools } = await getRouteAndGasQuote(
         network,
         collateral.symbol,
-        marketId,
         collateralAmount,
-        pools
+        marketId
     );
-
-    // return price per unit
-    return daiAmount.dividedBy(collateralAmount);
+    if (!pools && route) {
+        const generatedPools = await routeToPool(network, route, collateral.symbol, fees);
+        const daiAmount = await convertCollateralToDaiUsingPool(
+            network,
+            collateral.symbol,
+            marketId,
+            collateralAmount,
+            generatedPools
+        );
+        return { price: daiAmount.dividedBy(collateralAmount), pools: generatedPools };
+    }
+    if (totalPrice && pools) {
+        return { price: totalPrice.dividedBy(collateralAmount), pools: pools };
```

Comment:
The only remaining problem I see now is that `totalPrice` here is not the adjusted price that takes exchange fees into account (so we can properly compare exchange prices between each other), but the normal one

Reply:
i observe that the autorouter returns 2 amounts: 

```ts
    /**
     * The quote for the swap.
     * For EXACT_IN swaps this will be an amount of token out.
     * For EXACT_OUT this will be an amount of token in.
     */
    quote: CurrencyAmount;
    /**
     * The quote adjusted for the estimated gas used by the swap.
     * This is computed by estimating the amount of gas used by the swap, converting
     * this estimate to be in terms of the quote token, and subtracting that from the quote.
     * i.e. quoteGasAdjusted = quote - estimatedGasUsedQuoteToken
     */
    quoteGasAdjusted: CurrencyAmount;
```

My understanding of the above does not combine well with your claim.

The documentation claims that quote is the amount of "token out". If the value does not account for exchange fees, then the amount out is different from `quote` and hence the documentation is not valid.

Did you perhaps mean the gas fees? should i use `quoteGasAdjusted` value instead of `quote`?